src/algebra/associated.lean

```diff
@@ -316,6 +316,15 @@ lemma prime_of_associated [comm_monoid_with_zero α] {p q : α} (h : p ~ᵤ q) (
     ⟨λ ⟨v, hv⟩, hp.not_unit ⟨v * u⁻¹, by simp [hv, hu.symm]⟩,
       hu ▸ by { simp [units.mul_right_dvd], intros a b, exact hp.div_or_div }⟩⟩
 
+lemma primes_associated_of_dvd {α} [comm_cancel_monoid_with_zero α] {p q: α}
+  (p_prime: prime p) (q_prime: prime q) (dvd: p ∣ q): associated p q :=
```

Comment:
Please follow the [code style guidelines](https://leanprover-community.github.io/contribute/style.html): we follow type theory convention and put spaces before and after colons. The name change I'm not going to be firm on, but it would be easier to find for me this way.
```suggestion
lemma associated_of_prime_of_dvd [comm_cancel_monoid_with_zero α] {p q : α}
  (p_prime : prime p) (q_prime : prime q) (dvd : p ∣ q) : associated p q :=
```

Reply:
I prefer the name that @Vierkantor suggests